src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -3718,5 +3718,118 @@ void verifyTypeParameterAndParameter(TypeParameterSyntax typeParameter, Paramete
                 Assert.Same(symbol, parameterSymbol.Type);
             }
         }
+
+        public sealed class ScriptGlobals
+        {
+            public int SomeGlobal => 42;
+        }
+
+        [Fact]
+        public void CanAccessScriptGlobalsFromInsideMethod()
+        {
+            var source = @"
+void Method()
+{
+    LocalFunction();
+    void LocalFunction()
+    {
+        _ = SomeGlobal;
+    }
+}";
+            CreateSubmission(source, new[] { ScriptTestFixtures.HostRef }, hostObjectType: typeof(ScriptGlobals))
+                .VerifyEmitDiagnostics();
+        }
```

Comment:
I'm not very familiar with scripting tests. I assume we have a way of executing and verifying the console output. If so, could you adjust the tests to output something and verify it? Thanks

Reply:
Sure, I can try. `VerifyEmitDiagnostics` ensures that the script compiles without errors or warnings which addresses the issue I logged. If I remember correctly, I was mimicking other script compilation tests which only went this far. Beyond that I'm not sure what to test; just return the value of `SomeGlobal` and verify that it is 42?

Reply:
Yes, you could simply do `System.Console.Write(++SomeGlobal);` and check `43` is printed.
For running, looking at scripts tests in `CodeGenScriptTests.cs` and `ScriptSemanticTests.cs`, you can just use `CompileAndVerify(..., expectedOutput: ...)` as with non-script compilations.


---
In reply to: [175861703](https://github.com/dotnet/roslyn/pull/24921#discussion_r175861703) [](ancestors = 175861703)

Reply:
@jcouv `CompileAndVerify` requires the compilation to be an executable and `CreateSubmission` requires the compilation to be a DLL. I don't see a way to use a global host object with `CompileAndVerify`. The cross-submission tests are the ones I emulated here.

Reply:
There doesn't seem to be a way to pass host object instance to `CompileAndVerify`:

https://github.com/dotnet/roslyn/blob/f4d08be5b9b90457cf7a9fc1a2a2a628bcd72252/src/Test/Utilities/Portable/CompilationVerifier.cs#L96

Would you please take a look and let me know what I'm missing?

Reply:
Tagging @tmat for advice on testing script execution.

---
In reply to: [175964097](https://github.com/dotnet/roslyn/pull/24921#discussion_r175964097) [](ancestors = 175964097)

Reply:
@jnm2 How about the following to create a script:
`            var script = CSharpScript.Create<int>("X + Y", globalsType: typeof(Globals));` 
and then `script.EvaluateAsync().Result` (as in `PrivateTopLevel` test)?

---
In reply to: [176251542](https://github.com/dotnet/roslyn/pull/24921#discussion_r176251542) [](ancestors = 176251542,175964097)

Reply:
@jcouv Then we're [back full circle](https://github.com/dotnet/roslyn/pull/24921#discussion_r168986489) to where I was when @tmat asked me not to depend on the project containing `CSharpScript`.

Reply:
Hum. Ok.
@tmat How do you recommend we test the execution of those new script scenarios?

Reply:
@jcouv Write another test in the scripting test layer.

Reply:
I don't think we need to duplicate all the tests. The script running test is more like end-to-end test, so it can cover just a single scenario that involves local functions. Doesn't need to cover all the combinations that we already verified here that produced correct code.

Reply:
So one evaulation test for the global and one for a previous submission method?

Reply:
Yeah, it could be in a single test that has a few submissions.